nixos/modules/virtualisation/virtualbox-guest.nix

```diff
@@ -9,6 +9,28 @@ let
   cfg = config.virtualisation.virtualbox.guest;
   kernel = config.boot.kernelPackages;
 
+  mkVirtualBoxUserService = serviceArgs: {
+    description = "VirtualBox Guest User Services ${serviceArgs}";
+
+    wantedBy = [ "graphical-session.target" ];
+    partOf = [ "graphical-session.target" ];
+
+    # The graphical session may not be ready when starting the service
+    # Hence, check if the DISPLAY env var is set, otherwise fail, wait and retry again
+    startLimitBurst = 20;
+
+    unitConfig.ConditionVirtualization = cfg.conditionVirtualization;
+
+    # Check if the display environment is ready, otherwise fail
+    preStart = "${pkgs.bash}/bin/bash -c \"if [ -z $DISPLAY ]; then exit 1; fi\"";
```

Comment:
This should really not be necessary? graphical-session.target should be reached only after DISPLAY is set.

Reply:
Unfortunately it is. When testing it happened quite often that the graphical session was not yet ready. Or at least not in a state that the guest additions could start.

Reply:
Do you have a way to reproduce this? If this happens, it's indicative of a deeper ordering issue somewhere.

Reply:
Well, I started a VM. And on some starts the error occurred, on some it did not. Hence, I wrote this check. Since the check virtualbox starts as expected every time.

This should not be a blocker for this PR, right?